dev/bots/check_tests_cross_imports.dart

```diff
@@ -0,0 +1,572 @@
+    'packages/flutter/test/cupertino/text_field_test.dart',
+    'packages/flutter/test/cupertino/dialog_test.dart',
+    'packages/flutter/test/cupertino/date_picker_test.dart',
+    'packages/flutter/test/cupertino/switch_test.dart',
+    'packages/flutter/test/cupertino/magnifier_test.dart',
+    'packages/flutter/test/cupertino/text_field_restoration_test.dart',
+    'packages/flutter/test/cupertino/sheet_test.dart',
+    'packages/flutter/test/cupertino/action_sheet_test.dart',
+    'packages/flutter/test/cupertino/form_row_test.dart',
+    'packages/flutter/test/cupertino/colors_test.dart',
+    'packages/flutter/test/cupertino/text_form_field_row_restoration_test.dart',
+    'packages/flutter/test/cupertino/slider_test.dart',
+  };
+
+  static final Set<String> _knownCrossImports = knownWidgetsCrossImports.union(
+    knownCupertinoCrossImports,
+  );
+
+  /// Returns the Set of paths in `knownPaths` that are not in `files`.
+  static Set<String> _differencePaths(Set<String> knownPaths, Set<File> files) {
+    final Set<String> testPaths = files.map((File file) {
+      final int index = file.absolute.path.indexOf('packages/flutter/test');
+      final indexNormalized = index == -1 ? 0 : index;
+      return file.absolute.path.substring(indexNormalized);
+    }).toSet();
+    return knownPaths.difference(testPaths);
+  }
+
+  /// Returns a list of files in the given directory optionally matching the
+  /// given filenamePattern.
+  static List<File> _getFiles(Directory directory, [Pattern? filenamePattern]) {
+    final List<File> files = directory
+        .listSync(recursive: true)
+        .map((FileSystemEntity entity) {
+          if (entity is File) {
+            return entity;
+          } else {
+            return null;
+          }
+        })
+        .where((File? file) {
+          if (file == null) {
+            return false;
+          }
+          if (filenamePattern == null) {
+            return true;
+          }
+          return file.absolute.path.contains(filenamePattern);
+        })
+        .map<File>((File? s) => s!)
+        .toList();
+    return files;
+  }
+
+  static Set<File> _getUnknowns(Set<File> files, Set<String> knownPaths) {
+    return files.where((File file) {
+      final int index = file.absolute.path.indexOf('packages/flutter/test');
+      final String comparablePath = file.absolute.path.substring(index == -1 ? 0 : index);
+      return !knownPaths.contains(comparablePath);
+    }).toSet();
+  }
+
+  // Get a list of all the filenames in the source directory that end in "_test.dart".
+  static Set<File> _getTestFiles(Directory directory, _Library library) {
+    return _getFiles(directory.childDirectory(library.directory), RegExp(r'_test\.dart$')).toSet();
+  }
+
+  /// Returns true only if the file imports the given Library.
+  static bool _containsImport(File testFile, _Library library) {
+    final String contents = testFile.readAsStringSync();
+    return contents.contains(library.regExp);
```

Comment:
How long does it take to run this script? (If it's slow, we can likely speed it up by only reading each test file once. But I wouldn't bother with this optimization unless it's necessary)

Reply:
Ok gotcha, that seems fine 👍